src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -269,17 +269,7 @@ internal virtual void Validate()
                 throw new InvalidOperationException(CoreErrorMessages.NoClientIdWasSpecified);
             }
 
-            if (!Guid.TryParse(Config.ClientId, out Guid clientIdGuid))
-            {
-                throw new InvalidOperationException(CoreErrorMessages.ClientIdMustBeAGuid);
-            }
```

Comment:
Why remove this? What value will it have in ADFS? 

Aside: if truly not a Guid for ADFS then perhaps it's better to say if !adfs instead of removing it.